packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -191,14 +197,28 @@ class _FlutterPlatform extends PlatformPlugin {
       ));
 
       // Start the engine subprocess.
-      printTrace('test $ourTestCount: starting shell process');
+      printTrace('test $ourTestCount: starting shell process${previewDart2? " in preview-dart-2 mode":""}');
+
+      final String mainDart = previewDart2
+        ? await compile(
+            sdkRoot: artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath),
+            incrementalCompilerByteStorePath: '' /* not null is enough */,
+            mainPath: listenerFile.path,
```

Comment:
also add parameter strongMode: true

Reply:
The test fails in strongMode, we won't be able to confirm the flow until flutter and frontend is strong mode clean. Given that I would prefer to add strong mode separately in follow up PR.

Reply:
Ok as discussed off line you will land this PR, follow it up with a PR to make it work on linux too and another PR to add the strong mode option.
It is important to get the strong mode option is as we want to get to the errors quickly.